include/rapidcheck/detail/BitStream.hpp

```diff
@@ -40,6 +40,7 @@ template <typename Source>
 template <typename T>
 T BitStream<Source>::next(int nbits, std::false_type) {
   using SourceType = decltype(m_source.next());
+  SourceType sourceTypeLength = (sizeof(SourceType) * 8);
```

Comment:
Actually, you even have `numBits<T>()` defined above which is specifically for this purpose. Regardless, the value is `constexpr`.